src/libs/actions/Transaction.js

```diff
@@ -87,6 +93,11 @@ function saveWaypoint(transactionID, index, waypoint) {
             },
         },
     });
+    const recentWaypointAlreadyExists = _.find(recentWaypoints, (recentWaypoint) => recentWaypoint.address === waypoint.address);
+    if (!recentWaypointAlreadyExists) {
+        recentWaypoints.unshift(waypoint);
+        Onyx.merge(ONYXKEYS.NVP_RECENT_WAYPOINTS, recentWaypoints.slice(0, 5));
```

Comment:
Not a bug, but I think we shouldn't modify `recentWaypoints`

Reply:
That's a good point. I didn't think about this modifying the reference. I will change this.